File: array_part.py
# Time complexity - O(nlogn) 
# Space Complexity: O(1) 
# Did this code successfully run on LeetCode?: Yes
# Problems faced while coding this:None
# Approach : 1. Sort the array. 2.add every number at odd position to the sum.

class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        if nums is None or len(nums)==0:
            return 0
        sum = 0
        nums.sort()
        for i in range(0,len(nums),2):
            sum += nums[i]
            
        return sum

File: ArrayPart.java
import java.util.*;

class Solution {
    // Tc: O(nlogn) Sc: O(1)
    public int arrayPairSum(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);
        int sum = 0;
        for (int i = 0; i < n; i += 2) {
            sum += nums[i];
        }
        return sum;
    }
}